build/build.sh

```diff
@@ -17,19 +16,31 @@ echo "ce-build-revision:${REVISION}"
 echo "ce-build-output:${OUTPUT}"
 
 PREFIX_DIR=/opt/compiler-explorer/vala-${VERSION}
+mkdir -p "${PREFIX_DIR}"
+
+# "install" libraries and development headers required by the compiler to the prefix
+apt-get update && apt-get download libglib2.0-0 libglib2.0-dev libpcre3 libpcre3-dev
+find . -type f -name "*.deb" -exec dpkg-deb -x {} "${PREFIX_DIR}" \;
+
+# Strip the /usr prefix from the installed directory structure
+cp -R "${PREFIX_DIR}/usr/"* "${PREFIX_DIR}"
+rm -rf "${PREFIX_DIR:?}/usr"
```

Comment:
why the `:?` here?

Reply:
actually, why the `rm` ? :)

Reply:
Ah - sorry I see what it's doing now, I think. I see you're re-rooting the contents of `usr/...` but you don't say why?

Reply:
Extracting the deb files results in a `usr/` folder inside the `/opt/compiler-explorer/...` prefix that contains all the installed libraries in `usr/lib`, `usr/share` etc...

Setting the prefix to `/opt/compiler-explorer/...` on vala while building it means that vala things get installed directly into `lib`, `share` etc subfolders.

So to get the version of Vala we're building and the libraries we're getting from the package manager in the same directory structure, we strip the `/usr` prefix

Reply:
> why the `:?` here?

Sorry, didn't realise you were still waiting on an answer for this.

This will error out if the variable is not defined. I always do it out of habit for safety in commands involving `rm` 😅

It can be removed if you wish.

Reply:
Got it: it ewas just jarring to see it used in the lines above without a `:?` as well as directly beneath the block that set the vraiable: I wanted to understand the specifics! Thanks for explaining why in this case :)

Reply:
Got it: it ewas just jarring to see it used in the lines above without a `:?` as well as directly beneath the block that set the vraiable: I wanted to understand the specifics! Thanks for explaining why in this case :)